### ISSUE TITLE
App::Helpers

### ISSUE THREAD

**Issue by samseaver** [ModelSEED/ModelSEED#127]
I'm a little confused about the use of App::Helpers, so far the functions I've seen used are functions that can be used by other objects, or so it seems to me.  In other words, what's the different between:

```
my ($biochemistry,$ref) = $helper->get_object("biochemistry", $args, $store);
```

and

```
my $biochemistry = $store->get_object($biochemistry_reference);
```

**Comment by devoid** [comment]
Yea the interfaces for many App::Helpers functions aren't very clear. But
the difference is that get_object function calls the get_base_ref function.
get_base_ref takes the same arguments but also consumes from STDIN if the
string "-" is included in the arguments. It also tries to use a value set
in the configuration file...

Should we do things this way? I'm not sure. But at the time it seemed
reasonable to support getting data from STDIN, despite all of the tricks
that involves.

**Comment by samseaver** [comment]
Oh I get it, I see it now, it's likely to be useful if one were to pass in a list of aliases etc. though I don't think many would use it.

**Comment by devoid** [comment]
Yup. And the whole "insert a `-` as a command argument " is really stupid because you want to just pipe the results from command to command. But it's tricky to know when it's safe to read from STDIN and there's the unclear case of what to do if you have BOTH arguments and STDIN.

I'm still not settled on this; if someone has a better, cleaner idea I'm all for it.

**Comment by cshenry** [comment]
I don't think we need to support piping. And there are definitely tweaks we could make to the CLI, but I'd like to spend more time actually using the system before settling on the final design.
